server/src/optima/project.py

```diff
@@ -317,13 +317,33 @@ def exportGraph():
     data = json.loads(request.data)
     name = data['name']
     filename = name+'.xlsx'
-    columns = data['columns']
+    sheet = [{
+        "columns":data['columns']
+    }]
     path = fullpath(filename)
-    table = OptimaGraphTable(name, columns)
+    table = OptimaGraphTable(name, sheet)
     table.create(path)
     (dirname, basename) = os.path.split(path)
     return helpers.send_file(path, mimetype='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet')
 
+"""
+saves All data as Excel files
+"""
+@project.route('/exportall', methods=['POST'])
+@login_required
+@report_exception()
+def exportAllGraphs():
+    from sim.makeworkbook import OptimaGraphTable
+    
+    data = json.loads(request.data)
+    project_name = request.headers['Project']
+    name = project_name + " Optimization analyses"
```

Comment:
@TizianoPerrucci I have fixed this. Actually this path is for storing temporary files only. The file that is downloaded uses the correct name depending on the screen you are on.

See line 379 on  client/source/js/modules/common/save-graph-as-directive.js